sdk/core/Azure.Core.Experimental/src/MutableJsonElement.cs

```diff
@@ -121,18 +121,32 @@ public double GetDouble()
 
             if (Changes.TryGetChange(_path, _highWaterMark, out MutableJsonChange change))
             {
-                switch (change.Value)
+                return change.Value switch
                 {
-                    case double d:
-                        return d;
-                    case JsonElement element:
-                        return element.GetDouble();
-                    default:
-                        throw new InvalidOperationException($"Element at {_path} is not a double.");
-                }
+                    double d => d,
+                    JsonElement element => element.GetDouble(),
+                    _ => throw new InvalidCastException(GetInvalidCastExceptionText(typeof(double), _path, change)),
+                };
             }
 
-            return _element.GetDouble();
+            try
+            {
+                return _element.GetDouble();
+            }
+            catch (InvalidOperationException e)
+            {
+                throw new InvalidCastException(GetInvalidCastExceptionText(typeof(double), _path, _element), e);
+            }
+        }
+
+        private static string GetInvalidCastExceptionText(Type target, string path, MutableJsonChange change)
+        {
+            return $"Unable to get element as {target}.  Element at {path} has type {change.Value?.GetType()}.";
```

Comment:
If `Value` can be null, should we put it in quotes or similar to avoid the odd message when we don't have one?

`Unable to get element as SomeType.  Element at SomePath has type."`